File: views/src/main/java/dlgdev/weighttracker/dagger/NavigationControllerModule.java
package dlgdev.weighttracker.dagger;

import android.support.v7.app.AppCompatActivity;

import dagger.Module;
import dagger.Provides;
import dlgdev.weighttracker.domain.NavigationController;
import dlgdev.weighttracker.views.AppNavigation;

@Module
public class NavigationControllerModule {
	private AppCompatActivity activity;

	public NavigationControllerModule(AppCompatActivity activity) {
		this.activity = activity;
	}

	@Provides NavigationController provideNavigationController(AppNavigation navigation) {
		return navigation;
	}

	@Provides AppCompatActivity provideActivity() {
		return activity;
	}
}

File: views/src/main/java/dlgdev/weighttracker/dagger/NavigationControllerModule.kt
package dlgdev.weighttracker.dagger

import android.support.v7.app.AppCompatActivity

import dagger.Module
import dagger.Provides
import dlgdev.weighttracker.domain.NavigationController
import dlgdev.weighttracker.views.AppNavigation

@Module
class NavigationControllerModule(private val activity: AppCompatActivity) {

    @Provides internal fun provideNavigationController(navigation: AppNavigation): NavigationController {
        return navigation
    }

    @Provides internal fun provideActivity(): AppCompatActivity {
        return activity
    }
}
